### PR TITLE
Fix Intel warnings (closes #284)

### LINKED ISSUE #284: Cluster compiler warning templated matrix and tensors
>warning #3466: inheriting constructors must be inherited from a direct base class
      using Eigen::Tensor<Scalar, Rank>::Tensor;  // inherit base constructors
>warning #3466: inheriting constructors must be inherited from a direct base class
      using Eigen::Matrix<Scalar, Rows, Cols>::Matrix;  // inherit base constructors

Known warning, does not appear to have adverse effects.

### PR DESCRIPTION
**Pull request by guacke**

### CODE REVIEW

**Comment by lelemmen** [comment]
Thanks for fixing this. Could you try to see if changing

```cpp
enum {
    Rows = _Rows,
    Cols = _Cols
};
```

to

```cpp
static constexpr auto Rows = _Rows;
static constexpr auto Cols = _Cols;
```

and then using `Rows` and `Cols` again inside the `using` construct also produces the warning?

**Comment by tmhuysen** [comment]
Using the `static constexpr` also works, no warnings when compiling on the Linux with Intel compilers.

File: include/math/Matrix.hpp [hunk 1]
```diff
@@ -69,7 +66,7 @@ class Matrix : public Eigen::Matrix<_Scalar, _Rows, _Cols> {
      *  CONSTRUCTORS
      */
 
-    using Eigen::Matrix<Scalar, Rows, Cols>::Matrix;  // inherit base constructors
+    using Eigen::Matrix<_Scalar, _Rows, _Cols>::Matrix;  // inherit base constructors
```
Comment (lelemmen) [inline include/math/Matrix.hpp]:
```suggestion
    using Eigen::Matrix<Scalar, Rows, Cols>::Matrix;  // inherit base constructors
```

File: include/math/Tensor.hpp [hunk 2]
```diff
@@ -53,7 +50,7 @@ class Tensor : public Eigen::Tensor<_Scalar, _Rank> {
      *  CONSTRUCTORS
      */
 
-    using Eigen::Tensor<Scalar, Rank>::Tensor;  // inherit base constructors
+    using Eigen::Tensor<Scalar, _Rank>::Tensor;  // inherit base constructors
```
Comment (lelemmen) [inline include/math/Tensor.hpp]:
```suggestion
    using Eigen::Tensor<Scalar, Rank>::Tensor;  // inherit base constructors
```

**Comment by lelemmen** [comment]
@tmhuysen It seems like this PR is getting stale: is there anything that remains to be done?

File: include/math/Tensor.hpp (hunk 2 above)
Comment (tmhuysen) [reply include/math/Tensor.hpp]: underscores appear to be required to prevent the warnings.

**Pull request merged**